src/target/riscv/debug_reg_printer.c

```diff
@@ -62,22 +62,35 @@ static uint64_t riscv_debug_reg_field_value(riscv_debug_reg_field_info_t field,
 
 static unsigned int riscv_debug_reg_fields_to_s(char *buf, unsigned int offset,
 	struct riscv_debug_reg_field_list_t (*get_next)(riscv_debug_reg_ctx_t contex),
-	riscv_debug_reg_ctx_t context, uint64_t value)
+	riscv_debug_reg_ctx_t context, uint64_t value,
+	enum riscv_debug_reg_show show)
 {
 	unsigned int curr = offset;
-	curr += get_len_or_sprintf(buf, curr, " { ");
+	curr += get_len_or_sprintf(buf, curr, " {");
+	char *separator = "";
 	for (struct riscv_debug_reg_field_list_t list; get_next; get_next = list.get_next) {
 		list = get_next(context);
-		curr += riscv_debug_reg_field_to_s(buf, curr, list.field, context,
-				riscv_debug_reg_field_value(list.field, value));
-		curr += get_len_or_sprintf(buf, curr, ", ");
+
+		uint64_t field_value = riscv_debug_reg_field_value(list.field, value);
+
+		if (show == RISCV_DEBUG_REG_SHOW_ALL ||
+				(show == RISCV_DEBUG_REG_HIDE_UNNAMED_0 &&
+					(field_value != 0 ||
+						(list.field.values && list.field.values[0]))) ||
+				(show == RISCV_DEBUG_REG_HIDE_ALL_0 && field_value != 0)) {
+			curr += get_len_or_sprintf(buf, curr, separator);
+			curr += riscv_debug_reg_field_to_s(buf, curr, list.field, context,
+							field_value);
+			separator = " ";
+		}
```

Comment:
The condition of the "if" is complex, which makes it hard to understand and modify. Please, could you simplify this whole part of the code - for example as shown below.

```suggestion

		uint64_t field_value = riscv_debug_reg_field_value(list.field, value);

		const bool isZero = (field_value == 0);
		const bool isUnnamed = (!list.field.values) || (!list.field.values[0]);		
		
		if (display_mode == RISCV_DEBUG_REG_HIDE_UNNAMED_0 && isUnnamed && isZero)
			// Don't display this field
			continue;
			
		if (display_mode == RISCV_DEBUG_REG_HIDE_ALL_0 && isZero)
			// Don't display this field
			continue;			
		
		// Show this field
		curr += get_len_or_sprintf(buf, curr, separator);
		curr += riscv_debug_reg_field_to_s(buf, curr, list.field, context,
							field_value);
		separator = " ";		
		
```

Reply:
Same as above: This change should go to the debug spec repository instead (riscv-debug-spec).